deployment_role_bootstrap/role.tf

```diff
@@ -80,71 +80,253 @@ resource "aws_iam_policy" "deployment_role_sml_policy" {
+                  "arn:aws:lambda:eu-west-2:115311790871:function:dev-alerter",
+                  "arn:aws:lambda:eu-west-2:115311790871:function:preprod-healthcheck",
+                  "arn:aws:lambda:eu-west-2:115311790871:function:preprod-alerter",
+                  "arn:aws:lambda:eu-west-2:115311790871:function:prod-healthcheck",
+                  "arn:aws:lambda:eu-west-2:115311790871:function:prod-alerter"
+              ]
+            },
+            {
+              "Sid": "AllowCloudwatchEventsAccess",
+              "Effect": "Allow",
+              "Action": [
+                "events:PutRule",
+                "events:DescribeRule",
+                "events:DeleteRule",
+                "events:ListRules",
+                "events:ListRuleNamesByTarget",
+                "events:ListTagsForResource",
+                "events:PutTargets",
+                "events:ListTargetsByRule",
+                "events:RemoveTargets"
+              ],
+              "Resource": [
+                  "arn:aws:events:eu-west-2:115311790871:rule/dev-sml.aws.onsdigital.uk-healthcheck-trigger",
+                  "arn:aws:events:eu-west-2:115311790871:rule/preprod-sml.aws.onsdigital.uk-healthcheck-trigger",
+                  "arn:aws:events:eu-west-2:115311790871:rule/statisticalmethodslibrary.ons.gov.uk"
+              ]
+            },
+            {
+              "Sid": "AllowSNSAccess",
+              "Effect": "Allow",
+              "Action": [
+                "sns:ListTopics",
+                "sns:SetTopicAttributes",
+                "sns:GetTopicAttributes",
+                "sns:DeleteTopic",
+                "sns:Publish",
+                "sns:ListTagsForResource",
+                "sns:TagResource",
+                "sns:Subscribe",
+                "sns:Unsubscribe",
+                "sns:GetSubscriptionAttributes",
+                "sns:CreateTopic"
+              ],
+              "Resource": "arn:aws:sns:eu-west-2:115311790871:smlPortalTopic"
+            },
+            {
+              "Sid": "AllowIamLambdaAccess",
+              "Effect": "Allow",
+              "Action": [
+                "iam:PassRole",
+                "iam:CreateRole",
+                "iam:DeleteRole",
+                "iam:GetRole",
+                "iam:DetachRolePolicy",
+                "iam:UpdateAssumeRolePolicy",
+                "iam:ListRolePolicies",
+                "iam:ListInstanceProfilesForRole",
+                "iam:CreatePolicyVersion",
+                "iam:DeletePolicyVersion",
+                "iam:ListAttachedRolePolicies",
+                "iam:CreatePolicy",
+                "iam:GetPolicy",
+                "iam:ListPolicyTags",
+                "iam:GetGroupPolicy",
+                "iam:GetPolicyVersion",
+                "iam:GetRolePolicy",
+                "iam:DeletePolicy",
+                "iam:ListPolicyVersions",
+                "iam:AttachRolePolicy"
+              ],
+              "Resource": [
```

Comment:
This may not reflect whats actually in AWS but should the resources not be seperate per instance? would having preprod and prod resources in dev not potentially cause issues?

Reply:
I will generalise it now :)